Sources/SwiftUI/ImageBinder.swift

```diff
@@ -102,21 +103,17 @@ extension KFImage {
                         switch result {
                         case .success(let value):
                             self.loadedImage = value.image
-                            let r = RetrieveImageResult(
-                                image: value.image, cacheType: value.cacheType, source: value.source, originalSource: value.originalSource
-                            )
-                            CallbackQueue.mainCurrentOrAsync.execute {
-                                done(.success(r))
-                            }
+
+                            self.isLoaded.wrappedValue = true
+
+                            let animation = self.fadeTransitionDuration(cacheType: value.cacheType)
+                                .map { duration in Animation.linear(duration: duration) }
+                            withAnimation(animation) { self.loaded = true }
 
```

Comment:
shouldn't this block be called from the `mainCurrentOrAsync` callback queue ?

Reply:
Yes. You are right!